### PR TITLE
Field helpers

### PR DESCRIPTION
**Pull request by mwielocha**
Where this ugly thing here:

```scala
object VenueClaimR extends RCcMeta[VenueClaim]("venueclaims") {

  val venueid = new ObjectIdField("vid", this)
  val status = new EnumField[ClaimStatus.type, VenueClaimR.type]("status", this)
  val reason = new OptEnumField[RejectReason.type, VenueClaimR.type]("reason", this)
  val userId = new LongField("userId", this)
}
```

Becomes less ugly this:

```scala
class VenueClaimRMeta extends RCcMetaExt[VenueClaim, VenueClaimRMeta]("venueclaims") {

  val venueid = ObjectIdField("vid")
  val status = EnumField[ClaimStatus]
  val reason = OptEnumField[RejectReason]
  val userId = LongField
}
 
val VenueClaimR = new VenueClaimRMeta
```

### CODE REVIEW

**Comment by jakubmarton** [comment]
`val venueid = new ObjectIdField("vid", this)`
vs
`val venueid = ObjectIdField`

How will this field be named in db when there is no explicit name? Will it be `venueid`?

**Comment by mwielocha** [comment]
@jakubmarton yes, thats half of the fun, its quite complicated to do, but it works, just needs some extra testing, check out `resolve` method in `QueryFieldHelpers` to see how this is implemented, I added some comments there, since the code is not trivial unfortunately.

**Comment by mwielocha** [comment]
Of course you can provide a different name, I see now that `venueid` was actually `vid`, so I updated the description.

**Comment by mwielocha** [comment]
Also notice that we no longer use objects since objects are actually an inherited class which makes the type inference more complicated and not really needed.

**Comment by mwielocha** [comment]
It is now also possible to compose meta classes from traits like so:

File: cc/src/main/scala/me/sgrouples/rogue/cc/QueryFieldHelpers.scala [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+import scala.reflect.runtime.universe._
+import shapeless.tag
+import shapeless.tag._
+
+private[cc] sealed trait Marker
+
+trait QueryFieldHelpers[Meta] extends {
+  requires: Meta =>
+
+  private val names: mutable.Map[Int, String] = mutable.Map.empty
```
Comment (marekzebrowski) [inline cc/src/main/scala/me/sgrouples/rogue/cc/QueryFieldHelpers.scala]: `private[this]` for names, fields, nameId ?

Comment (mwielocha) [inline cc/src/main/scala/me/sgrouples/rogue/cc/QueryFieldHelpers.scala]: Isn't `[this]` the default...?

Comment (marekzebrowski) [inline cc/src/main/scala/me/sgrouples/rogue/cc/QueryFieldHelpers.scala]: unfortunately no:
http://stackoverflow.com/questions/9698677/privatethis-vs-private

Comment (mwielocha) [inline cc/src/main/scala/me/sgrouples/rogue/cc/QueryFieldHelpers.scala]: I see, sure I can change that.

**Pull request merged**